### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by antrix1989**
## Proposed changes

Fix warnings found by the static analyzer.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/src/MSIDError.h [hunk 1]
```diff
@@ -21,15 +21,15 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
```
Comment (kaisong1990) [inline IdentityCore/src/MSIDError.h:21]: This pull request does not update changelog.txt.

Please consider if this change would be noticeable to a partner or user and either update changelog.txt or resolve this conversation.

File: IdentityCore/src/broker_operation/response/MSIDBrokerNativeAppOperationResponse.m [hunk 2]
```diff
@@ -67,7 +67,7 @@ + (NSString *)responseType
 
 - (NSNumber *)httpStatusCode
 {
-    if (!_httpStatusCode) _httpStatusCode = @200;
+    if (_httpStatusCode == nil) _httpStatusCode = @200;
```
Comment (oldalton) [inline IdentityCore/src/broker_operation/response/MSIDBrokerNativeAppOperationResponse.m:70]: nit: do we really want to do `== nil`? Any way to suppress this warning?

File: IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m [hunk 3]
```diff
@@ -47,7 +47,9 @@ - (instancetype)initWithCoder:(NSCoder *)decoder
         self.apiId = (NSInteger)[decoder decodeFloatForKey:kApiId];
         
         NSString *uuIdString = [decoder decodeObjectForKey:kCorrelationID];
-        self.correlationId = ![NSString msidIsStringNilOrBlank:uuIdString] ? [[NSUUID UUID] initWithUUIDString:uuIdString] : nil;
+        if ([NSString msidIsStringNilOrBlank:uuIdString]) return nil;
```
Comment (oldalton) [inline IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m:50]: looks like a change of behavior. Before we would just set correlationId to nil if uuIdString is nil. Now we return nil. Is this expected?

**Review by oldalton — APPROVED** [review]
Added a few comments. Please take a look before merging.

File: IdentityCore/src/broker_operation/response/MSIDBrokerNativeAppOperationResponse.m (hunk 2 above)
Comment (antrix1989) [reply IdentityCore/src/broker_operation/response/MSIDBrokerNativeAppOperationResponse.m]: We don't need to follow this pattern for other cases, seems anylizer throws this error only for NSNumbers:

`Converting a pointer value of type 'NSNumber * _Nonnull' to a primitive boolean value; instead, either compare the pointer to nil or call -boolValue`

File: IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m (hunk 3 above)
Comment (antrix1989) [reply IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m:50]: fixed

File: IdentityCore/src/auth_scheme/MSIDAuthenticationScheme.h [hunk 4]
```diff
@@ -43,7 +43,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, readonly) MSIDAuthScheme authScheme;
 @property (nonatomic, readonly) NSDictionary *schemeParameters;
 @property (nonatomic, readonly) MSIDCredentialType credentialType;
-@property (nonatomic, readonly) NSString *tokenType;
+@property (nonatomic, nullable, readonly) NSString *tokenType;
```
Comment (kaisong1990) [inline IdentityCore/src/auth_scheme/MSIDAuthenticationScheme.h:46]: can ```tokenType``` be nullable?

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m [hunk 5]
```diff
@@ -299,6 +299,8 @@ - (void)logAndFillError:(NSString *)errorTitle status:(OSStatus)status error:(NS
     {
         *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, description, nil, nil, nil, nil, nil, NO);
     }
+    
+    return YES;
```
Comment (kaisong1990) [inline IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:303]: Does it always return ```YES```?

File: IdentityCore/src/auth_scheme/MSIDAuthenticationScheme.h (hunk 4 above)
Comment (antrix1989) [reply IdentityCore/src/auth_scheme/MSIDAuthenticationScheme.h:46]: yes, it is nullable for the Bearer type.

File: IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m (hunk 5 above)
Comment (antrix1989) [reply IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m:303]: yes, the analyzer complained about Apple code style: the method that fills error should return a boolean that indicates whether error was filled or not.

**Pull request merged**